cupy/core/fusion.py

```diff
@@ -829,29 +925,42 @@ def _create_ufunc(cupy_ufunc, numpy_ufunc):
 
 class reduction(object):
 
-    def __init__(self, cupy_op, numpy_op):
+    def __init__(self, cupy_op, numpy_op, raw):
         self._cupy_op = cupy_op
         self._numpy_op = numpy_op
+        self._raw = raw
         self.__doc__ = cupy_op.__doc__
+        self.identity = raw.identity
+        self._preamble = raw._preamble
 
     def __call__(self, *args, **kwargs):
+        arg = args[0]
+        if isinstance(arg, FusionVarPython):
+            if len(args) != 1:
+                raise Exception("Can't reduce a tuple")
+            dtype = arg.dtype
+            history = arg._history
+            for op in self._raw._ops:
+                if numpy.can_cast(dtype.type, op[0][0]):
+                    return_type = numpy.dtype(op[1][0])
```

Comment:
Can you improve readability of these indexing on `op`? (at least comments?)

Reply:
Avoided indexing.